### PR TITLE
Add hack/update-test-owners.sh and update test_owners.csv

### PR DESCRIPTION
**Pull request by dims**
Manual update to test_owners.csv, Add a update script that
simply calls the python script

Note that verify-test-owners.sh does not fail when things are out
of whack (details in PR #36096)

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/36663)

### CODE REVIEW

**Comment by dims** [comment]
@saad-ali : test_owners.cvs is out of date, but we don't fail on it. Should we fix this for 1.5?

File: hack/verify-test-owners.sh [hunk 1]
```diff
@@ -23,7 +23,7 @@ source "${KUBE_ROOT}/hack/lib/init.sh"
 
 cd "${KUBE_ROOT}"
 if ! hack/update_owners.py --check; then
-    echo 'Run ./hack/update_owners.py to fix it'
+    echo 'Run ./hack/update-test-owners.py to fix it'
```
Comment (saad-ali) [inline hack/verify-test-owners.sh]: `.py` -> `.sh`

File: hack/update-test-owners.sh [hunk 2]
```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env bash
```
Comment (saad-ali) [inline hack/update-test-owners.sh]: Why the wrapper shell script? Nice to have?

**Review by saad-ali — CHANGES_REQUESTED** [review]
Thanks for fixing!

File: hack/verify-test-owners.sh (hunk 1 above)
Comment (dims) [inline hack/verify-test-owners.sh]: Done

File: hack/update-test-owners.sh (hunk 2 above)
Comment (dims) [inline hack/update-test-owners.sh]: fixed. just a relic from the file i copied

File: hack/verify-test-owners.sh [hunk 3]
```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/env bash
+#!/bin/bash
```
Comment (ixdy) [inline hack/verify-test-owners.sh]: please revert the change to this line

**Review by ixdy — COMMENTED** [review]
I agree with @saad-ali - it's not clear to me why we need a no-op wrapper shell script around the python script.

File: hack/verify-test-owners.sh (hunk 3 above)
Comment (dims) [inline hack/verify-test-owners.sh]: @ixdy the main reqadon was, when i looked at the directory, it looked obvious that there were various update-*.sh scripts that updated different things and there was this solitary update_owners.py which did not appear to me as one of them. 

Also, since the python script runs go commands, i thought adding KUBE_ROOT would be prudent as well just like the other scripts. hence the wrapper script.

Comment (rmmh) [inline hack/verify-test-owners.sh]: We have 378 uses of `#!/bin/bash`, and 26 uses of `#!/usr/bin/env bash`. I think assuming bash is /bin/bash is safe.
